xarray/core/rolling.py

```diff
@@ -162,9 +167,13 @@ def reduce(self, func, **kwargs):
             dims.remove(dims[self._axis_num])
             windows = [DataArray(window, dims=dims) for window in windows]
```

Comment:
Just a note -- I probably should have caught this earlier -- but it might be cleaner to use the `reduce` method on each window instead, e.g.,
```
windows = [window.reduce(func, dims=self.dims, **kwargs) for _, window in self]
```

That would let you skip this logic to recreate the DataArrays -- and would also generalize immediately to Dataset.